cliTest/testcommandlineparser.cpp

```diff
@@ -210,6 +210,15 @@ TEST_CASE("Run Tests","Check we only run if it's appropriate"){
         REQUIRE_THROWS_WITH(cmdP.run(perfSink.get()), Catch::Contains("Trying to run with invalid command line parameters"));
     }
 
+    SECTION("Fail run with version on the command line")
+    {
+        CommandLineParser cmdP(factoryMock.get());
+        ArgumentHolder ah{"prog", "-v", "-m", "TEST1", "-f", "inputfile.graph", "-o", "outputfile.graph"};
+        cmdP.parse(ah.argc(), ah.argv());
+        REQUIRE_FALSE(cmdP.isValid());
+        REQUIRE_THROWS_WITH(cmdP.run(perfSink.get()), Catch::Contains("Trying to run with invalid command line parameters"));
+    }
+
```

Comment:
What happens if `-v` is specified at the end of an otherwise valid command line?